src/main/java/de/tum/in/www1/artemis/config/migration/MigrationService.java

```diff
@@ -0,0 +1,158 @@
+     *
+     * @param event         Specifies when this method gets called and provides the event with all application data
+     * @param entryClassMap The changelog to be executed
+     */
+    public void execute(ApplicationReadyEvent event, SortedMap<Integer, Class<? extends MigrationEntry>> entryClassMap) {
+        if (event.getApplicationContext().getEnvironment().acceptsProfiles(Profiles.of(SPRING_PROFILE_TEST))) {
+            return;
+        }
+
+        log.info("Starting Artemis migration");
+
+        SortedMap<Integer, MigrationEntry> entryMap = instantiateEntryMap(entryClassMap);
+
+        if (!checkIntegrity(entryMap)) {
+            log.error("{} corrupted. Aborting startup.", getClass().getSimpleName());
+            event.getApplicationContext().close();
+            System.exit(1);
+        }
+        else {
+            log.info("Integrity check passed.");
+        }
+
+        List<String> executedChanges = migrationChangeRepository.findAll().stream().map(MigrationChangelog::getDateString).toList();
+
+        Map<Integer, MigrationEntry> migrationEntryMap = entryMap.entrySet().stream().filter(entry -> !executedChanges.contains(entry.getValue().date()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        if (!migrationEntryMap.isEmpty()) {
+            log.info("Executing migration entries");
+            String startupHash = toMD5(ZonedDateTime.now().toString());
+            for (Map.Entry<Integer, MigrationEntry> integerClassEntry : migrationEntryMap.entrySet()) {
+                MigrationEntry entry = integerClassEntry.getValue();
+                log.debug("Executing entry {}", entry.date());
+                entry.execute();
+                MigrationChangelog newChangelog = new MigrationChangelog();
+                newChangelog.setAuthor(entry.author());
+                newChangelog.setDateExecuted(ZonedDateTime.now());
+                newChangelog.setDateString(entry.date());
+                newChangelog.setSystemVersion(artemisVersion);
+                newChangelog.setDeploymentId(startupHash);
+
+                migrationChangeRepository.save(newChangelog);
+                log.debug("Done");
+            }
+            log.info("Executed {} migration entries", migrationEntryMap.size());
+        }
+        else {
+            log.info("No migration entries executed");
+        }
+        log.info("Ending Artemis migration");
+    }
+
+    public SortedMap<Integer, MigrationEntry> instantiateEntryMap(SortedMap<Integer, Class<? extends MigrationEntry>> entryClassMap) {
+        return entryClassMap.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, entry ->
+        // We have to manually autowire the components here
+        (MigrationEntry) beanFactory.autowire(entry.getValue(), AutowireCapableBeanFactory.AUTOWIRE_CONSTRUCTOR, true), (prev, next) -> next, TreeMap::new));
```

Comment:
```suggestion
        return entryClassMap.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, entry -> {
            // We have to manually autowire the components here
            return (MigrationEntry) beanFactory.autowire(entry.getValue(), AutowireCapableBeanFactory.AUTOWIRE_CONSTRUCTOR, true);
        }, (prev, next) -> next, TreeMap::new));
```
I think due to the comment the automatic formatting is a bit weird here. Either the suggestion or moving the comment to the line before the return should probably yield better results.

Reply:
fixed